block/manager.go

```diff
@@ -291,7 +301,10 @@ func (m *Manager) SyncLoop(ctx context.Context, cancel context.CancelFunc) {
 				"daHeight", daHeight,
 				"hash", block.Hash(),
 			)
-			m.syncCache[block.SignedHeader.Header.BaseHeader.Height] = block
+			if m.syncCache[block.SignedHeader.Header.BaseHeader.Height] == nil {
```

Comment:
`height := block.SignedHeader.Header.BaseHeader.Height` and use `height` in multiple places